### DISCUSSION TITLE
Architekt Azure potrzebny od zaraz ;-)

### DISCUSSION THREAD

**Discussion by serverlesspolska** [AzureCommunityPL/group#192] [Oferty pracy]
Cześć,

W moim projekcie AWSowym dodajemy chmurę Azure. Potrzebny nam architekt, który ogarnie wszystko od początku do konca na Azure:
- architektura rozwiązania (sieci wirtualne, VMki + komponent aplikacyjny serverless)
- proces devops w Azure DevOps
- wszystko IaC (ARM)
- znajomość IoT Hub mile widziana

Po projekcie (min. 6 miesiecy) kontynuacja wspólpracy na innych projektach, u innych klientów.

💰 **Wynagrodzenie**:  B2B 140 - 180 PLN netto na godzinę.💰

Więcej szczegółów na priv.

**Comment by aryll** [comment]
bardzo mi się podoba stwierdzenie  🤣
`proces devops w Azure DevOps`

**Reply by serverlesspolska** [reply]
Może brzmi śmiesznie ale obecny projekt używa BitBucket i usług AWS.

Dla Azurowca niekoniecznie to byłby plus, dlatego zaznaczam, że dla ułatwienia życia taką decyzję podjąłem, że rzeczy dot. Azure mają tam być ogarnięte i jest pełna dowolność jak. Byle porządnie 😉

**Reply by eklime** [reply]
To ja tylko tak "rzucę", że jedno to architektura, ale klepanie IaC, ARM i Pipelines to raczej robota dla inżyniera ;) czyli potrzebujesz dwóch osób

**Reply by kaluzaaa** [reply]
@eklime raczej jedna wystarczy. Chyba, że to architekt od pptx :-)

**Reply by serverlesspolska** [reply]
no ja sie zgadzam z @kaluzaaa. Jeśli ktoś chce tylko rysować prostokąty w swojej wieży z kości słoniowej (a takich wielu spotkałem w swoim życiu) to mu proponuje młodzieżowe słowo roku 2020: #wypier...ać :-)

Przez lata mojej pracy z AWS wiem, że jak ktoś się nie zajmuje implementacją to 💩  wie. A takich ludzi to ja na projekcie nie potrzebuje 😀

**Reply by aryll** [reply]
a czy to nie jest tak że rola architekta to ta mocno niedoprecyzowana i to co taki człowiek robi (co wie i jak się tam znalazł) w dużej mierze zależy od firmy?

czyli architekt (ten co wie i zna) architektowi (ten co dostał awans bo siedzi w firmie) nierówny? 😄

**Reply by serverlesspolska** [reply]
@aryll być może role architekta nawet nie jest aż tak niedopracowana.  

Natomiast moje rozumienie (i w firmie dla której pracuje) jest takie, że _hands-on-experiance_ jest wymagane. I jak trzeba zejść do kodu to się  schodzi do kodu, a nie że sobie garnitur ubrudzę... 🤣

EDIT:
Przy okazji jest jeden programista z wiedzą dużą o Azure, więc nim trzeba tez zarządzać i dawać zadania, ale praca techniczna dla architekta tez się znajdzie.

**Reply by aryll** [reply]
gorzej jak ktoś nie nosi garnituru 🤣

ja się zgadzam że trzeba umieć zrobić prostokąty które się rysuje - umieć a robić na co dzień to dwie różne rzeczy i to chyba zależy od firmy i wielkości rozwiązań ;) 

traktujcie ten mój przerywnik jako "nie znam się ale się wypowiem" 🤣

**Reply by maciejgos** [reply]
zgadzam się z @aryll, oraz @serverlesspolska oraz sam wychodzę z założenia, że umiem zrobić prostokąty które narysowałem 😉

**Reply by eklime** [reply]
> zgadzam się z @aryll, oraz @serverlesspolska oraz sam wychodzę z założenia, że umiem zrobić prostokąty które narysowałem 😉

Ja też umiem, ale to nie znaczy, że lubię klepać pajplajny w DevOps - od tego jest inżynier, a nie architekt.
Hands on experience to jedno, ale klepanie armow na etacie to drugie ;)
Ja na codzień sobie coś tam grzebię w Azure, ale jak jest zadanie przygotowania infrastruktury pod nową apke, to robię to za pomocą zadań dla zespołu inżynierów z wytycznymi i wskazówkami.

**Reply by piotrgwiazda** [reply]
Według mojego wewnętrznego kompasu to jest zapotrzebowanie na "Inżynier Chmurowy/DevOps"

Z ciekawostek - u nas architekt ma rozumieć jak rozwiązać problem instycji finansowej, rozmawia z CTO/CIO i innymi C3PO w bankach, i w zadaniach architekta jest zaprojektowanie przepływów danych, messagingu vs event-drive arch, wykorzystania hurtowni gdzie trzeba, a gdize nie trzeba to Sparków itp., podział frontendów, architektura mikrousług, proces wdrażania i oczywiście zaplanowanie pipelineów w DevOps. Jako jedno z zadań jest narysowanie szczegółów w jakiejś chmurze a potem IaaC, zaplanowanie testowania infrastruktury, compliance as code itp. Ja mam zawsze otwarte przynajmniej kilka projektów w Terraform. (Nigdize nie korzystamy z ARM, nie ma na to czasu :-) ) i architekt musi mieć o tym pojęcie.

DevOps/Cloud  Engineer też umie narysować prostokąty w chmurze i musi umieć je wdrożyć w Terraform wraz z zespołem. Może nie znać się na DDD, mikroserwisach, data gridach i analityce danych w Sparku, ale musi umieć w pipeline'y, security, testowanie infry itp. Odcinek od projektu chmury do wdrożeina na prod. 

Nie mamy roli "rysowacza prostokątów chmurowych" bez części solution arch, albo bez części DevOps. Nie mamy takiej roli.

**Reply by aryll** [reply]
a tak ciągnąc ten temat: w firmach, które znacie kto (albo w jakisposób) jest odpowiedzialny za komunikację między zespołami z technicznego punktu widzenia? (np. zespół "frontent" ma używać API zespołu "backend" i muszą się dogadać)

tak, wiem że wątek tej konkretnej oferty nie jest najlepszy na te rozmowy ale w sumie jestem ciekawa Waszego doświadczenia w tym temacie 😄 🛩️

**Reply by serverlesspolska** [reply]
@piotrgwiazda jak zwał tak zwał ;-)

Dodam tylko, że u mnie na projekcie codziennie się rozmawia z klientem, natomiast cel jest już od dawna ustalony i ponieważ to jest dodatek do AWS, to pracy analitycznej jest tu bardzo mało.

Gdy projekt się skończy, to będą inni klienci i inne zadania do zrealizowania. Wtedy też umiejętności, które opisałeś, w tym spotkania z klientami i projektowanie rozwiązań wychodząc od potrzeb będą miały miejsce. (Wydaje mi się, że to szeroko rozumiane odpowiedzialności Solution Architect'a).

To, co Ty opisałeś to też mi trochę zahacza o bycie koordynatorem technicznym kilku projektów. 

U nas są oczywiście osoby w zespołach, co realizują zadania programistyczne/techniczne, ale raczej wolimy czystą sytuację: ktoś projektuje, robi, wdraża i bierze odpowiedzialność od początku do końca za swoją (i zespołu) pracę. Przynajmniej jak tak chce aby było u mnie na projekcie, u innych klientów to juz nie wnikam :-P

**Reply by serverlesspolska** [reply]
@aryll im starszy jestem tym bardziej anarchistą jestem w tym temacie - mały zespół, co robi wszystko od początku do końca i to jeszcze ogarnia operacyjnie. No ale to trzeba mieć serverless do tego :-)

Im mniej jednostek organizacyjnych tym mniej strat na komunikacji i przepychanek my-wy-oni. Two pizza size team i robimy wszystko sami w oderwaniu od reszty firmy. 

Lubię ten model bardzo, bo moim zdaniem to najlepsza droga do szybkiego i kreatywnego tworzenia rozwiązań, innowacji. 
I nie marnuje życia na a) głupie spotkania b) przepychanki c) inne głupoty i niekompetencje ludzi. Ten brak marnowania życia to dla mnie bardzo ważny aspekt !

Żeby była jasność te +/- 8 osób to mamy ze mną w zespole u klienta. Klient to ogromna firma, ale my jesteśmy odseparowani od reszty jego firmy. 

BTW pracodawca z ogłoszenia to duża, polska firma -> https://ttpsc.com/pl/ - żeby nie było że do startupu Was zapraszam ;-)

**Reply by aryll** [reply]
@serverlesspolska faktycznie, nie wszyscy uwielbiają spotkania bez celu i sensu 😉

**Reply by serverlesspolska** [reply]
@aryll dlatego ja wole unikać spotkań z klientami na korzyść "dłubania w chmurze" i dostarczania rozwiązań. Kiedyś pracowałem jako Analityk Biznesowy i swoje juz wyrobiłem ;-)

**Reply by piotrgwiazda** [reply]
> a tak ciągnąc ten temat: w firmach, które znacie kto (albo w jakisposób) jest odpowiedzialny za komunikację między zespołami z technicznego punktu widzenia? (np. zespół "frontent" ma używać API zespołu "backend" i muszą się dogadać)
> 
> tak, wiem że wątek tej konkretnej oferty nie jest najlepszy na te rozmowy ale w sumie jestem ciekawa Waszego doświadczenia w tym temacie 😄 🛩️

Rozdzielanie front/back jest szkodliwe. Międzymordzie w postaci kogoś do koordynacji to strata czasu. Chyba, że podejście API first gdzie mamy contract testing (Pact/Spring Cloud  Contract) i front. Tyle, że front to wtedy zwykle JS i jakiś BFF. 

Co do reszty rozmowy to DevOpsi faktycznie cierpliwie "dłubią w chmurze", robią w Terraformie i w pajplajnach unikając spotkań z klientem. Mało to jednak związane z archiketurą rozwiązań.

Natomiast zespoły dostarczające kawałek systemu w całości to dobre podejście. Minimum koordynacji "powyżej" i mikrousługi poniżej. Zgadzam się w 100% i nie ma w tym nic z anarchii. Trudno jest jednak zrobić to w oderwaniu od organizacji klienta jeśli to jakaś główna domena biznesowa.

**Comment by serverlesspolska** [comment]
Pytanie konkretne: czy ktoś jest zainteresowany powyższym stanowiskiem?
